### ISSUE TITLE
Make Value updateable post initialization

### ISSUE THREAD

**Issue by jon49** [ardalis/SmartEnum#17]
I would like to do something like this:

```csharp
    public class MessageHandler : SmartEnum<MessageHandler, Action<string>>
    {

        protected MessageHandler(string name, Action<string> value) : base(name, value)
        {
        }

        public void Update(Action<string> value)
        {
            this.Value = value;
        }

        public static MessageHandler ToBeDefinedLater =
            new MessageHandler(nameof(ToBeDefinedLater), _ => { });
    }
```

Which would be called like this:

```csharp
MessageHandler.ToBeDefinedLater.Update(s => { "ugh" });
```

This would allow me to define MessageHandler in one project use it in another and inject the item I would like to work with from the parent project without having to share resources.

I'm not sure if this would cause any problems. But it would definitely make like simpler!

If you know of any workarounds that would be nice. I'm still working on figure it out.

I enjoy your podcast by-the-way. That is how I heard of your library!

**Comment by jon49** [comment]
For now I'll do something like this (not tested). But maybe making `Value` `protected set;` would fix the problem?

```csharp
typeof(MessageHandler).GetField(nameof(Value), BindingFlags.Instance | BindingFlags.NonPublic).SetValue(this, value);
```

When I get time I'll look into it. Assuming you are OK making that change? Otherwise I won't take the time :-)

**Comment by ardalis** [comment]
Hi @jon49 ! Glad to hear you're a listener on WeeklyDevTips.com. The idea for this library is to essentially replace enums with something more extensible. Up until now, I haven't thought about allowing for the value to change, since that seems like to breaks one of the basic invariants of enums. On the other hand, it *is* a way to extend behavior beyond what `enum` allows. I don't think it would cause any problems to make Value protected set, potentially. If you want to do a quick PR I'd certainly consider it.

**Comment by jon49** [comment]
I ended up tackling the problem differently. But if I find I need this capability again I will definitely do a PR!

**Comment by ardalis** [comment]
Done and available in v1.05 on Nuget.
